wireshark-*: pass in overridden libpcap in all-packages.nix

Izorkin opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
Build wireshark-cli wothout bluetooth

cc @justin-gerhardt @bjornfor
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

justin-gerhardt commented:
It seems the root issue is the bluez occasionally errors on build per https://github.com/NixOS/nixpkgs/issues/91011. As far as I can tell this is nothing specifically to do with wireshare itself. If bluez is to be removed as a dependency it doesn't really make sense to do that to the cli version for wireshark but not the QT version. Personally, given the 9 out of 1000 failure rate I don't think this is warrented.

Izorkin commented:
My build bluez always freezing.

Izorkin commented:
On servers are unlikely to used bluetooth.

Izorkin commented:
cc @Mic92 @flokli @aanderse

flokli commented on pkgs/top-level/all-packages.nix:
```diff
@@ -20462,12 +20462,16 @@ in
-  wireshark-cli = wireshark.override { withQt = false; };
+  wireshark-cli = wireshark.override {
+    withQt = false;
+    libpcap = libpcap.override { withBluez = false; };
```
```suggestion
    libpcap = libpcap.override { withBluez = stdenv.isLinux; };
```

Izorkin replied:
Thanks, update.

flokli reviewed (changes requested):
As written in https://github.com/NixOS/nixpkgs/issues/91011, bluez seems a bit flaky in general. We shouldn't try to patch this around in other things consuming bluez, but fix it properly.

Please move the `libpcap` override to `all-packages.nix` (as you did), but keep with `withBluez = stdenv.isLinux;` as is.

You can then easily override `wireshark-cli` in your server overlay.

People might want to still have `wireshark-cli` with bluez support.

Izorkin committed:
wireshark-*: pass in overridden libpcap in all-packages.nix

This allows building wireshark-* with a explicitly disabled bluezSupport
in an overlay.

flokli commented:
The commit message is still the old one (and we don't disable bluetooth anymore). Maybe

> wireshark-*: pass in overridden libpcap in all-packages.nix
>
> This allows building wireshark-* with a explicitly disabled bluezSupport in an overlay.

Izorkin commented:
Thanks, update.
